src/unit.c

```diff
@@ -1764,6 +1778,7 @@ SEXP multUnit(SEXP unit, double value) {
 	return mult;
 }
 SEXP multUnits(SEXP units, SEXP values) {
+    if (!inherits(units, "unit_v2")) error(_("old version of unit class is no longer allowed"));
```

Comment:
This appears inconsistent with addUnits().  There is the same check in addUnits(), but I believe that check is necessary because Ops.unit() (in R code) does not do a check on the argument to multUnits(), BUT Ops.unit() calls as.unit() on the arguments to addUnits().

So ...

EITHER add an as.unit() in R code on the argument to multUnits() and remove this check in multUnits() AND the check addUnits()

OR leave this check in multUnits() and the check in addUnits() and remove the as.unit() calls in Ops.unit()

Or am I missing something ?

Reply:
`as.unit()` is necessary to ensure that both arguments are not `simpleUnit` so it can't be removed.

I'll make all the check in R code then...